spec/jobs/movements_on_date_job_spec.rb

```diff
@@ -1,38 +1,76 @@
 require 'rails_helper'
 
 RSpec.describe MovementsOnDateJob, type: :job do
-  let!(:prison) { create(:prison, code: 'MDI') }
-  let(:nomis_offender_id) { 'G3462VT' }
-  let!(:alloc) { create(:allocation_history, nomis_offender_id: nomis_offender_id, secondary_pom_nomis_id: 123_435, prison: 'MDI') }
+  let(:nomis_offender_id) { nomis_offender.fetch(:offenderNo) }
+  let(:nomis_offender) { build(:nomis_offender, agencyId: to_prison_code) }
+  let(:case_info) { create(:case_information) }
+  let!(:offender_record) { create(:offender, nomis_offender_id: nomis_offender_id, case_information: case_info) }
+  let(:today) { Time.zone.today }
+  let(:yesterday) { today - 1.day }
+  let(:pom) { build(:pom) }
+  let!(:allocation) { create(:allocation_history, nomis_offender_id: nomis_offender_id, primary_pom_nomis_id: pom.staff_id, prison: from_prison_code) }
+  let(:mail_params) {
+    {
+      email: pom.email_address,
+      pom_name: pom.first_name.titleize,
+      offender_name: "#{nomis_offender.fetch(:lastName)}, #{nomis_offender.fetch(:firstName)}".titleize,
+      nomis_offender_id: nomis_offender_id,
+      prison_name: Prison.find(from_prison_code).name,
+      url: "http://localhost:3000/prisons/#{from_prison_code}/staff/#{pom.staff_id}/caseload"
+    }
+  }
 
   before do
-    stub_auth_token
-  end
+    stub_offender(nomis_offender)
+    stub_poms(from_prison_code, [pom])
+
+    # Stub offender movements from yesterday
+    stub_request(:get, "#{ApiHelper::T3}/movements?fromDateTime=#{yesterday - 1.year}T00:00&movementDate=#{yesterday}").
+      to_return(body: [movement].to_json)
 
-  it 'deallocates' do
-    allow(OffenderService).to receive(:get_offender).and_return(build(:hmpps_api_offender))
```

Comment:
When the new offender wrapper was introduced and therefore the bug too, then this test didn't catch it? Is that why you have deleted it? It is stubbing OffenderService and returning the hmpps_api_offender so therefore hiding the bug?

Reply:
Yep – that's correct! The behaviour of the `OffenderService.get_offender` method (and the type of object returned) changed. But since it was stubbed it out here, it didn't reflect the new behaviour.

The method used to return a `HmppsApi::Offender` object – which is why this stub returned:
```ruby
build(:hmpps_api_offender)
```

However it now returns an MpcOffender object, which is a wrapper around that & other records.

The two methods have a similar interface, but very subtly different meanings/behaviour.